crates/subspace-gateway/Cargo.toml

```diff
@@ -0,0 +1,36 @@
+[package]
+name = "subspace-gateway"
+version = "0.1.0"
+authors = ["Teor <teor@riseup.net>"]
+description = "A Subspace Network data gateway."
+edition = "2021"
+license = "GPL-3.0-or-later"
+build = "build.rs"
+homepage = "https://subspace.network"
+repository = "https://github.com/autonomys/subspace"
+include = [
+    "/src",
+    "/build.rs",
+    "/Cargo.toml",
+    "/README.md"
+]
+
+[package.metadata.docs.rs]
+targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+anyhow = "1.0.89"
+clap = { version = "4.5.18", features = ["derive"] }
+fdlimit = "0.3.0"
+futures = "0.3.30"
+mimalloc = "0.1.43"
+sc-cli = { git = "https://github.com/subspace/polkadot-sdk", rev = "5871818e1d736f1843eb9078f886290695165c42", default-features = false }
+sc-service = { git = "https://github.com/subspace/polkadot-sdk", rev = "5871818e1d736f1843eb9078f886290695165c42", default-features = false }
```

Comment:
I would very strongly prefer to not pull anything from Substrate in non-blockchain code. Just like farmer doesn't depend directly or indirectly on polkadot-sdk, this shouldn't either.

`sc_cli::Signals` for graceful shutdown can be replaced with relatively trivial implementation (farmer has one). 

`sc_service::TaskManager` is also Substrate-specific thing, we get away without it in farmer by combining futures explicitly and we also have `AsyncJoinOnDrop` to have guarantee that after dropping resources are in fact released (in contrast to Substrate, see https://github.com/autonomys/subspace/issues/578